conf/cmake/common.cmake

```diff
@@ -9,76 +9,87 @@ set(AREG_AR          "${CrossCompile}ar")
     set(AREG_BINARY "shared")
+else()
+    set(AREG_BINARY "static")
 endif()
 
-
 # Add compiler flags here
-add_compile_options(-g -pthread -Werror)
-
 # Checking Compiler for adding corresponded tweaks and flags
 if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
     # Clang compile options
-    add_compile_options(-stdlib=libc++)
+    list(APPEND CompileOptions -g -pthread -Werror -stdlib=libc++ ${UserDefines})
+    if (Config MATCHES "Release")
+        list(APPEND CompileOptions -O2)
+    endif()
+
 elseif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     # GNU compile options
-    add_compile_options(-Wall ${UserDefines})
+    list(APPEND CompileOptions -g -pthread -Werror -Wall ${UserDefines})
+    if (Config MATCHES "Release")
+        list(APPEND CompileOptions -O2)
+    endif()
+
 elseif (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
     # Visual Studio C++
+    if(Config MATCHES "Release")
+        add_definitions(-DNDEBUG -DUNICODE)
+    else()
+        list(APPEND CompileOptions -Od -RTC1)
+        add_definitions(-DDEBUG -DUNICODE)
+    endif()
+
 endif()
 
 
 if(Config MATCHES "Release")
-    add_compile_options(-O2)
-    add_definitions(-DNDEBUG)
+    add_definitions(-DNDEBUG )
 else()
     add_definitions(-DDEBUG)
 endif()
 
 # flags for bitness
 if(Platform MATCHES "x86_64")
-    if(NOT DEFINED CrossCompile)
+    if(NOT DEFINED CrossCompile AND NOT AREG_OS MATCHES "Windows")
         if(bit MATCHES "32")
-            add_compile_options(-m32)
+            list(APPEND CompileOptions -m32)
         else()
-            add_compile_options(-m64)
+            list(APPEND CompileOptions -m64)
         endif()
     endif()
 endif()
 
 if(AREG_OS MATCHES "Windows")
 # Windows
     add_definitions(-DWINDOWS)
-    set(OBJ_EXT "obj")
-    set(AREG_BIN_EXT ".exe")
-    set(AREG_LIB_PREFIX)
-    set(AREG_STATIC_LIB ".lib")
-    if(AREG_BINARY MATCHES "shared")
-        set(AREG_LIB_EXT ".dll")
-    else()
-        set(AREG_LIB_EXT ".lib")
-    endif()
 else()
 # Unix
     add_definitions(-DPOSIX)
-    set(OBJ_EXT "o")
-    set(AREG_BIN_EXT ".out")
-    set(AREG_LIB_PREFIX "lib")
-    set(AREG_STATIC_LIB ".a")
-    if(AREG_BINARY MATCHES "shared")
-        set(AREG_LIB_EXT ".so")
-    else()
-        set(AREG_LIB_EXT ".a")
-    endif()
+endif()
+
+# Adding compile options
+add_compile_options(${CompileOptions})
+
+# Examples Compile Options
+if (NOT CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+    set(ExampleCompileOptShared "-Wl,-rpath=${AREG_OUTPUT_BIN} -L ${AREG_OUTPUT_BIN}")
+    set(ExampleCompileOptStatic "-L ${AREG_OUTPUT_LIB} -Wl,-Bstatic -Wl,-Bdynamic")
 endif()
 
 # Examples LD flags (-l is not necessary)
-list(APPEND exampleLDFlags areg m  stdc++ rt pthread)
+if(AREG_OS MATCHES "Windows")
+    list(APPEND LDFLAGS ws2_32 psapi)
+    list(APPEND exampleLDFlags areg ${LDFLAGS})
+else()
+    list(APPEND LDFLAGS m  stdc++ rt pthread)
+    list(APPEND exampleLDFlags areg ${LDFLAGS})
+endif()
+
+# CPP standard for examples
 set(exampleCXXStandard "17")
```

Comment:
All the projects must be compiled against the same standard, so defining a variable here for examples is not really needed, or maybe make it a more generic variable and use it for the whole source base.

Reply:
Has been resolved!